### ISSUE TITLE
Review why the API and GUI are displaying old logs (reading from AS pkl apparently)

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#1485]
Hello @dbeltrankyl, @LuiggiTenorioK 

### Autosubmit Version

4.1.11

### Expid affected(If applicable)

`t01v` in ClimateDT

### Which task has issues? Where is the log(If applicable)

`t01v_19900101_fc0_1_1_DN`

### Summary

`t01v` task `t01v_19900101_fc0_1_1_DN` failed and the logs were not transferred from the remote HPC to the VM. The GUI displayed the run status fine (the dates displayed were correctly), however, when you clicked on the out and err logs, the information displayed appeared to be from a previous run.

In the discussion on Gitlab, it looks like API and GUI are reading this data from the AS data on disk. Maybe this is related to the bug when logs are not transferred, or some other situation where this could happen. 

This is a serious issue, especially for operators.

### Steps to reproduce

Not sure. We have a workflow that failed, and logs were not transferred, and with logs for previously run jobs successfully transferred.

### What is the current bug behavior?

Stale data displayed to users.

### What is the expected correct behavior?

Display only the logs for the current run id.

### Relevant logs and/or screenshots(if applicable)

(Paste any relevant logs - use code blocks (```) to format console output, logs, and code, as
it's very hard to read otherwise.)

### Any other relevant information(if applicable)

()

**Comment by kinow** [comment]
mentioned in issue digital-twins/de_340-2/workflow#755

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Dec 10, 2024, 15:25

Hello @kinow , 

I think we should be more practical for this one ( due deadlines)

* Merge everything pending related to zombies to master
* Dev version + tsuite 
* Monitor them with the GUI and see if it still happens

And from my side, I can launch some big dummy experiment in the climate-dt using all platforms ( local, lumi-ps, lumi-slurm , marenostrum5-ps, marenostrum5-slurm) and monitor it

If it doesn't happen with any experiment, we can calmly review why it could happen in 4.1.11. 

And if it still happens even with the logs retrieved, maybe the origin of the issue is other stuff non-related to the log recovery process code

**Comment by kinow** [comment]
Hi @dbeltrankyl 

>I think we should be more practical for this one ( due deadlines)
>
>* Merge everything pending related to zombies to master
>* Dev version + tsuite
>* Monitor them with the GUI and see if it still happens

This is what I had in mind too. +1

> And from my side, I can launch some big dummy experiment in the climate-dt using all platforms ( local, lumi-ps, lumi-slurm , marenostrum5-ps, marenostrum5-slurm) and monitor it

That'd be helpful!

>If it doesn't happen with any experiment, we can calmly review why it could happen in 4.1.11.

:thumbsup: 

>And if it still happens even with the logs retrieved, maybe the origin of the issue is other stuff non-related to the log recovery process code

Yup. This is an issue found in the testing suite and that I remembered about when talking with @youcheng-csc (Cheng You, ClimateDT operator) yesterday.

I assigned 4.1.12 as I believe 4.1.12 will be released somewhere in January (as we are giving this version to operations committee, so we must only release it once we are 100% sure everything was tested and is working), so we should have more time to test while preparing for the d-suite (i.e. not planning to start working on this yet).

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Dec 11, 2024, 12:09

Also related to https://github.com/BSC-ES/autosubmit-api/issues/154#issuecomment-2535539727

**Comment by kinow** [comment]
In GitLab by @LuiggiTenorioK on Dec 11, 2024, 13:10

mentioned in issue digital-twins/de_340-2/workflow#761

**Comment by dbeltrankyl** [comment]
I believe this still happens? fyi @kinow 

Moving it to 4.1.13

**Comment by kinow** [comment]
Yes, I had the link to this issue in gitlab in the testing suite weekly issue (under the "Please test this" section). I've updated it to point to this issue, and will update once I have more info about it. Thanks Dani!
